src/test/ui/generic-associated-types/impl_bounds.stderr

```diff
@@ -1,20 +1,22 @@
-error: `impl` associated type signature for `A` doesn't match `trait` associated type signature
-  --> $DIR/impl_bounds.rs:15:5
+error[E0310]: the parameter type `T` may not live long enough
```

Comment:
This is weird, lol, but seems more like a side-effect than _due_ to these changes. I bet if we did the same setup with methods it would report the same error.

Reply:
Shouldn't we complain here that the bounds have changed?  There should be something saying that the `Self: 'static` bound does not match the `Self: 'a` bound from the trait.

Reply:
It think the problem here is that we're not just comparing the bounds syntactically, but instead using the lifetime machinery. 

So "given T: 'a, does T: 'static" (or vice versa) ends up erroring out in a harder to understand way than just a trait bounds mismatch... lifetime bounds mismatch sometimes can't even point to a bound at all when it's, for example, implied by the self type of the impl.

Reply:
I'm a bit suprised. In the case of a method, we get the expected "impl has stricter requirements than trait" error here too: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=8dada6c436de5aadf8d5a4d2f18cfd7d

Reply:
ah I found it.

Reply:
Fixed in 2bbcdc73339c2b7c3e57c2fc0d3bc574945c0023, then I stacked another commit on top that combines the obligation cause codes so something like this is harder to miss in the future (and adjusts the wording a bit).